### PR TITLE
[MM-25163] Review SimulController

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

This is a general review of SimulController mainly focusing on:

- Tune action frequencies to go with the [upcoming](https://github.com/mattermost/mattermost-load-test-ng/pull/286) controllers distribution.
- Fixed a couple of latent errors.
- Added an action to search for channels.
- Improved fetching users statuses to better reflect the webapp's behaviour.

#### Ticket

https://mattermost.atlassian.net/browse/MM-25163

### CODE REVIEW

File: loadtest/store/memstore/store.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func (s *MemStore) UserForPost(postId string) (string, error) {
 	if post, ok := s.posts[postId]; ok {
 		return post.UserId, nil
 	}
-	return "", nil
+	return "", ErrPostNotFound
```
Comment (agnivade) [inline loadtest/store/memstore/store.go:241]: Should we also make the same change in `UserForPost` and `FileInfoForPost` as well ?

File: loadtest/control/simulcontroller/actions.go [hunk 2]
```diff
@@ -770,3 +812,59 @@ func createMessage(u user.User, channel *model.Channel, isReply bool) (string, e
...
+	}
+
+	channel, err := u.Store().RandomChannel(team.Id, store.SelectAny)
+	if errors.Is(err, memstore.ErrChannelStoreEmpty) {
+		return control.UserActionResponse{Info: "no channel to search"}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.EmulateUserTyping(channel.Name[:1+rand.Intn(4)], func(term string) control.UserActionResponse {
```
[the remark this answers is not in the document]
Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go:861]: Unfortunately we don't have that kind of data. You either bump the action frequency or the number of API calls you make which are tied to the number of characters the user types.
I'll take a close look at all these cases in https://mattermost.atlassian.net/browse/MM-24645.

File: loadtest/store/memstore/store.go (hunk 1 above)
Comment (streamer45) [reply loadtest/store/memstore/store.go:241]: We can certainly do that :+1:

File: loadtest/control/simulcontroller/actions.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (agnivade) [reply loadtest/control/simulcontroller/actions.go:863]: Sure, just a comment mentioning that it's arbitrary should clarify things to the reader.

File: loadtest/control/actions.go [hunk 3]
```diff
@@ -39,8 +40,10 @@ func SignUp(u user.User) UserActionResponse {
 	username := u.Store().Username()
 	password := u.Store().Password()
 
-	err := u.SignUp(email, username, password)
-	if err != nil {
+	if err := u.SignUp(email, username, password); err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			return UserActionResponse{Info: fmt.Sprintf("%s has already signed up", username)}
```
Comment (isacikgoz) [inline loadtest/control/actions.go:45]: 👌

File: loadtest/control/simulcontroller/actions.go [hunk 4]
```diff
@@ -613,8 +656,7 @@ func (c *SimulController) addReaction(u user.User) control.UserActionResponse {
 		}
 	}
 
-	err = u.SaveReaction(reaction)
-	if err != nil {
+	if u.SaveReaction(reaction); err != nil {
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply loadtest/control/simulcontroller/actions.go:661]: What is the magic here @streamer45?

[the remark this answers is not in the document]
Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go:661]: Went a little too far with compactness.

File: loadtest/control/actions.go (hunk 3 above)
Comment (streamer45) [reply loadtest/control/actions.go:45]: Yeah, couldn't think of anything better.

**Pull request merged**